queue/amqp.go

```diff
@@ -288,9 +288,9 @@ func (q *AMQPQueue) PublishDelayed(j *Job, delay time.Duration) error {
 	)
 }
 
-// RepublishBuried will republish in the main queue all the jobs that timed out without Ack
-// or were Rejected with requeue = False.
-func (q *AMQPQueue) RepublishBuried() error {
+// RepublishBuried will republish in the main queue those jobs that timed out without Ack
+// or were Rejected with requeue = False and and makes complay return true.
+func (q *AMQPQueue) RepublishBuried(comply RepublishConditionFunc) error {
```

Comment:
Maybe to do not break the API we should create a new method called `RepublishBuriedConditionally` with all that logic, and a `RepublishBuried` calling to `RepublishBuriedConditionally` with a function that accepts everything.

Reply:
Or accept a `...RepublishConditionFunc` which would allow several small validation functions and you can still call it as `RepublishBuried()` thus not breaking API compat